heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/MetricsCacheManager.java

```diff
@@ -0,0 +1,258 @@
+    LOG.info(String.format("Starting MetricsCache for topology %s with topologyId %s with "
+            + "MetricsCache Id %s, MericsCache Port: %d.",
+        topologyName, topologyId, metricsCacheMgrId, masterPort));
+
+    LOG.info("System Config: " + systemConfig);
+
+    //-----------------
+    // Populate the msConfig
+    MetricsSinksConfig sinksConfig = new MetricsSinksConfig(metricsSinksConfigFilename);
+
+    LOG.info("Sinks Config: " + sinksConfig.toString());
+
+    //----------------
+    Config.Builder config = Config.newBuilder()
+        .putAll(ClusterDefaults.getDefaults())
+        .putAll(ClusterDefaults.getSandboxDefaults())
+        .putAll(ClusterConfig.loadConfig(Defaults.heronSandboxHome(),
+            Defaults.heronSandboxConf(), "./release.yaml"))
+        .put(Keys.cluster(), cluster)
+        .put(Keys.role(), role)
+        .put(Keys.environ(), environ)
+        .put(Keys.topologyName(), topologyName)
+        .put(Keys.topologyId(), topologyId);
+    LOG.info("Config: " + config.build().toString());
+
+    Config configExpand = Config.expand(config.build());
+    LOG.info("Config: " + config.toString());
+
+    //-----------------
+    TopologyMaster.MetricsCacheLocation metricscacheLocation =
+        TopologyMaster.MetricsCacheLocation.newBuilder()
+            .setTopologyName(topologyName)
+            .setTopologyId(topologyId)
+            .setHost(InetAddress.getLocalHost().getHostName())
+            .setControllerPort(-1) // not used for metricscache
+            .setMasterPort(masterPort)
+            .setStatsPort(statsPort)
+            .build();
+
+    MetricsCacheManager metricsCacheManager = new MetricsCacheManager(
+        topologyName, METRICS_CACHE_HOST, masterPort, statsPort, metricsCacheMgrId,
+        systemConfig, sinksConfig, configExpand, metricscacheLocation);
+    metricsCacheManager.start();
+
+    LOG.info("Loops terminated. MetricsCache Manager exits.");
+  }
+
+  public void start() throws Exception {
+    // 1. Do prepare work
+    // create an instance of state manager
+    String statemgrClass = Context.stateManagerClass(config);
+    LOG.info("Context.stateManagerClass " + statemgrClass);
+    IStateManager statemgr;
+    try {
+      statemgr = ReflectionUtils.newInstance(statemgrClass);
+    } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
+      throw new Exception(String.format(
+          "Failed to instantiate state manager class '%s'",
+          statemgrClass), e);
+    }
+
+    // Put it in a try block so that we can always clean resources
+    try {
+      // initialize the statemgr
+      statemgr.initialize(config);
+
+      statemgr.setMetricsCacheLocation(metricscacheLocation, topologyName);
+      LOG.info("metricscacheLocation " + metricscacheLocation.toString());
+      LOG.info("topologyName " + topologyName.toString());
+
+      LOG.info("Starting Metrics Cache HTTP Server");
+      metricsCacheManagerHttpServer.start();
```

Comment:
it is a http server. we use java http server.
as far as i know, heron does not have http server helper